src/core_modules/capture-core/components/WidgetProfile/DataEntry/factory/EnrollmentFactory.js

```diff
@@ -0,0 +1,200 @@
+// @flow
+/* eslint-disable no-underscore-dangle */
+import i18n from '@dhis2/d2-i18n';
+import { capitalizeFirstLetter } from 'capture-core-utils/string/capitalizeFirstLetter';
+import type {
+    CachedProgram,
+    CachedProgramSection,
+    CachedProgramTrackedEntityAttribute,
+    CachedTrackedEntityAttribute,
+    CachedTrackedEntityType,
+    CachedOptionSet,
+} from '../../../../storageControllers/cache.types';
+import type { TrackedEntityType } from '../../../../metaData';
+import { Enrollment, RenderFoundation, Section } from '../../../../metaData';
+import { DataElementFactory } from './DataElementFactory';
+
+type ConstructorInput = {|
+    cachedTrackedEntityAttributes: Map<string, CachedTrackedEntityAttribute>,
+    cachedOptionSets: Map<string, CachedOptionSet>,
+    cachedTrackedEntityTypes: Map<string, CachedTrackedEntityType>,
+    trackedEntityTypeCollection: Map<string, TrackedEntityType>,
+|};
+
+export class EnrollmentFactory {
+    static _addLabels(enrollment: Enrollment, cachedProgram: CachedProgram) {
+        if (cachedProgram.enrollmentDateLabel) {
+            enrollment.enrollmentDateLabel = cachedProgram.enrollmentDateLabel;
+        }
+        if (cachedProgram.incidentDateLabel) {
+            enrollment.incidentDateLabel = cachedProgram.incidentDateLabel;
+        }
+    }
+
+    static _addFlags(enrollment: Enrollment, cachedProgram: CachedProgram) {
+        enrollment.allowFutureEnrollmentDate = cachedProgram.selectEnrollmentDatesInFuture;
+        enrollment.allowFutureIncidentDate = cachedProgram.selectIncidentDatesInFuture;
+        enrollment.showIncidentDate = cachedProgram.displayIncidentDate;
+    }
+
+    static _getFeatureType(cachedFeatureType: ?string) {
+        return cachedFeatureType ? capitalizeFirstLetter(cachedFeatureType.toLowerCase()) : 'None';
+    }
+
+    dataElementFactory: DataElementFactory;
+    trackedEntityTypeCollection: Map<string, TrackedEntityType>;
+    cachedTrackedEntityTypes: Map<string, CachedTrackedEntityType>;
+    constructor({
+        cachedTrackedEntityAttributes,
+        cachedOptionSets,
+        cachedTrackedEntityTypes,
+        trackedEntityTypeCollection,
+    }: ConstructorInput) {
+        this.trackedEntityTypeCollection = trackedEntityTypeCollection;
+        this.cachedTrackedEntityTypes = cachedTrackedEntityTypes;
+        this.dataElementFactory = new DataElementFactory({
+            cachedTrackedEntityAttributes,
+            cachedOptionSets,
+        });
+    }
+
+    _buildTetFeatureTypeField(trackedEntityTypeId: ?string) {
```

Comment:
same here is it `tei`?

Reply:
Same as above, I think `tet` is correctly used here because the code is about the `tracked entity type`. Thanks!